pkg/ccl/backupccl/manifest_handling.go

```diff
@@ -213,7 +218,7 @@ func readBackupManifest(
 	filename string,
 	encryption *jobspb.BackupEncryptionOptions,
 ) (BackupManifest, int64, error) {
-	r, err := exportStore.ReadFile(ctx, filename)
+	r, err := readLatestCheckpointFile(ctx, exportStore, filename)
```

Comment:
hmmm should we have a check here too as we do in writeBackupManifest that is if we are actually attempting to read the `BACKUP_MANIFEST` we don't look at checkpoint subdirectory? I almost wished the two didn't share a code path but with all the memory monitoring I wouldn't make that split in this PR.